tests/integration/pilot/locality/failover/failover_test.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package failover
+
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"regexp"
+	"testing"
+	"time"
+
+	"text/template"
+
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/apps"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/tests/integration/pilot/locality"
+)
+
+// This test allows for Locality Load Balancing Failover testing without needing Kube nodes in multiple regions.
+// We do this by overriding the calling service's locality with the istio-locality label and the serving
+// service's locality with a service entry.
+//
+// Failover is set in the mesh config as follows:
+// failover:
+// - from: region
+// 	 to: closeregion
+//
+// The created Service Entry for fake-(cds|eds)-external-service-12345.com points the domain at services
+// that exist internally in the mesh. In the Service Entry we set a non-existent service or IP to be in the same
+// locality as the caller (A). Service B to be in the primary failover locality and service C to be in the secondary failover locality.
+// For CDS, the endpoint pool for the service in the local locality is empty so it fails over to service B.
```

Comment:
So in the config dump service b has priority: 0, we don't actively failover?

Reply:
service b is failover priority 1